addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java

```diff
@@ -117,63 +118,119 @@ public String getZoneMinderId() {
+    public void onThingStatusChanged(ThingStatus thingStatus) {
+        if (thingStatus == ThingStatus.ONLINE) {
+            IZoneMinderConnectionHandler connection = null;
+            try {
+                connection = aquireSessionWait();
+                IZoneMinderMonitor monitor = ZoneMinderFactory.getMonitorProxy(connection, config.getZoneMinderId());
+                IMonitorDataGeneral monitorData = monitor.getMonitorData();
+
+                logger.debug("{}:    SourceType:         {}", getLogIdentifier(), monitorData.getSourceType().name());
+                logger.debug("{}:    Format:             {}", getLogIdentifier(), monitorData.getFormat());
+                logger.debug("{}:    AlarmFrameCount:    {}", getLogIdentifier(), monitorData.getAlarmFrameCount());
+                logger.debug("{}:    AlarmMaxFPS:        {}", getLogIdentifier(), monitorData.getAlarmMaxFPS());
+                logger.debug("{}:    AnalysisFPS:        {}", getLogIdentifier(), monitorData.getAnalysisFPS());
+                logger.debug("{}:    Height x Width:     {} x {}", getLogIdentifier(), monitorData.getHeight(),
+                        monitorData.getWidth());
+            } catch (ZoneMinderInvalidData | ZoneMinderAuthenticationException | ZoneMinderGeneralException
+                    | ZoneMinderResponseException ex) {
+                logger.error("{}: context='onThingStatusChanged' error in call to 'getMonitorData' - Message='{}'",
+                        getLogIdentifier(), ex.getMessage(), ex.getCause());
+
+            } finally {
+                if (connection != null) {
+                    releaseSession();
+                }
+            }
 
-            logger.info("{}: Unsubscribing from Monitor Events: {}", getLogIdentifier(),
-                    bridge.getThing().getUID().getAsString());
-            ZoneMinderFactory.UnsubscribeMonitorEvents(config.getZoneMinderId(), this);
+            try {
+                updateMonitorProperties();
 
-            logger.debug("{}: Calling parent onBridgeConnected()", getLogIdentifier());
-            super.onBridgeDisconnected(bridge);
+            } catch (Exception ex) {
+                logger.error(
+                        "{}: context='onThingStatusChanged' - Exception occurred when calling 'updateMonitorPropoerties()'. Exception='{}'",
+                        getLogIdentifier(), ex.getMessage());
+
+            }
+        }
+    }
 
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        try {
+            if (!channelUID.getId().equals(ZoneMinderConstants.CHANNEL_ONLINE)) {
+                dataConverter.subscribe(channelUID);
+            }
+            super.channelLinked(channelUID);
+
+            logger.info("{}: context='channelLinked' - Unlinking from channel '{}'", getLogIdentifier(),
+                    channelUID.getAsString());
         } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.